nemo/collections/tts/models/magpietts.py

```diff
@@ -2527,3 +2592,486 @@ def setup_test_data(self, dataset_cfg):
+                        )
+                        _attn_prior[bidx, 0, min(text_time_step_attended[bidx] - left_offset + 2, _text_len - 1)] = 0.8
+                        _attn_prior[bidx, 0, min(text_time_step_attended[bidx] - left_offset + 3, _text_len - 1) :] = (
+                            prior_epsilon * prior_epsilon
+                        )
+
+                # Penalize timesteps that have been attended to more than 10 times
+                for _timestep in attended_timestep_counter[bidx]:
+                    if _timestep > left_offset and attended_timestep_counter[bidx][_timestep] >= 10:
+                        # This means the timestep has been attended to more than 10 times (To avoid getting stuck)
+                        _attn_prior[bidx, 0, : _timestep - left_offset + 1] = prior_epsilon
+
+                if not end_of_text:
+                    unfinished_texts[bidx] = True
+                else:
+                    unfinished_texts[bidx] = False
+                    if text_time_step_attended[bidx] - left_offset < text_lens[bidx] - 3:
+                        # This means the sentence has definitely not ended
+                        if bidx not in end_indices:
+                            unfinished_texts[bidx] = True
+
+                if end_of_text and (
+                    text_time_step_attended[bidx] - left_offset >= text_lens[bidx] - 4 or bidx in end_indices
+                ):
+                    if bidx not in finished_texts_counter:
+                        finished_texts_counter[bidx] = 0
+
+                chunk_end_has_reached = False
+                if not end_of_text and text_time_step_attended[bidx] - left_offset >= text_lens[bidx] - 4:
+                    # This means entire text has not been provided yet and audio generation for current chunk has almost reached the end
+                    chunk_end_has_reached = True
+
+        return _attn_prior, unfinished_texts, finished_texts_counter, chunk_end_has_reached
+
+    def generate_speech_per_chunk_of_text(
+        self,
+        batch,
+        end_of_text,
+        beginning_of_text,
+        max_decoder_steps=500,
+        temperature=0.7,
+        topk=80,
+        use_cfg=True,
+        cfg_scale=1.0,
+        estimate_alignment_from_layers=None,
+        lookahead_window_size=5,
+        start_prior_after_n_audio_steps=10,
+        apply_attention_prior=False,
+        apply_prior_to_layers=None,
+        prior_epsilon=1e-5,
+        compute_all_heads_attn_maps=False,
+        return_cross_attn_probs=False,
+        use_exponential_weight=False,
+    ):
+        """
+        This method is used for streaming inference. Method to do text to speech inference one chunk of text/words at a time.
+        This method only works with batch size 1 right now. Before calling this method for the first time please call the method
+        set_streaming_inference_variables()
+        """
```

Comment:
Can you docstring the new args you added to this function that are not present in infer_batch?

Reply:
Done.